runtime/src/main/java/org/corfudb/runtime/LogReplicationClient.java

```diff
@@ -0,0 +1,277 @@
+        }
+    }
+
+    /**
+     * Add destination to LogReplicationSourceMetadataTable for monitoring
+     *
+     * @param logicalGroup String representative of the group of tables to be replicated
+     * @param destination Cluster which the logicalGroup should be replicated to
+     * @return true if destination was newly registered
+     */
+    public boolean addDestination(String logicalGroup, String destination) {
+        if (Strings.isNullOrEmpty(logicalGroup) || Strings.isNullOrEmpty(destination)) {
+            log.error("Invalid logicalGroup or destination");
+            return false;
+        }
+        try {
+            return IRetry.build(ExponentialBackoffRetry.class, () -> {
+                try (TxnContext txn = corfuStore.txn(CORFU_SYSTEM_NAMESPACE)) {
+
+                    ClientInfoKey clientInfoKey = ClientInfoKey.newBuilder()
+                            .setClientName(clientName)
+                            .setModel(LogReplication.ReplicationModel.valueOf(modelEnum))
+                            .setGroupName(logicalGroup)
+                            .build();
+
+                    SinksInfoVal clientDestinations;
+
+                    if (txn.isExists(LR_SOURCE_METADATA_TABLE_NAME, clientInfoKey)) {
+                        clientDestinations = txn.getRecord(sourceMetadataTable, clientInfoKey).getPayload();
+
+                        if (clientDestinations.getDestinationIdsList().contains(destination)) {
+                            log.warn("Destination already added");
+                            return false;
+                        }
+
+                        clientDestinations = clientDestinations.toBuilder().addDestinationIds(destination).build();
+                    }
+                    else {
+                        clientDestinations = SinksInfoVal.newBuilder().addDestinationIds(destination).build();
+                    }
+
+                    txn.putRecord(sourceMetadataTable, clientInfoKey, clientDestinations, null);
+                    txn.commit();
+
+                    return true;
+                }
+                catch (Exception e) {
+                    log.error("Unable to add destination", e);
+                    return false;
+                }
+            }).setOptions(x -> x.setMaxRetryThreshold(Duration.ofSeconds(SYNC_THRESHOLD))).run();
+        } catch (InterruptedException e) {
+            throw new UnrecoverableCorfuInterruptedError(e);
+        }
+    }
+
+    /**
+     * Add a list destination to LogReplicationSourceMetadataTable for monitoring
+     *
+     * @param logicalGroup String representative of the group of tables to be replicated
+     * @param remoteDestinations Clusters which the logicalGroup should be replicated to
+     * @return true if all destinations were newly registered
+     */
+    public boolean addDestination(String logicalGroup, List<String> remoteDestinations) {
+        if (Strings.isNullOrEmpty(logicalGroup)) {
```

Comment:
we should check remoteDestinations as well, right?

I think we should not allow clients to pass an empty/null list.